utils.go

```diff
@@ -722,22 +783,47 @@ func checkMntSrcOwnership(mntSrc string, uid, gid uint32) (bool, uint32, uint32,
 	mntSrcUid = st.Uid
 	mntSrcGid = st.Gid
 
-	if mntSrcUid != uid || mntSrcGid != gid {
+	// We expect the host uid assigned to the container to be equal or higher
+	// than the uid of the dir being mounted into the container (i.e., the former
+	// is a container-range uid, the latter is a uid in the normal range of host
+	// uids).
+
+	if uid > mntSrcUid && gid > mntSrcGid {
 		return true, mntSrcUid, mntSrcGid, nil
 	}
 
-	// mnt src subdir ownership check
+	if uid < mntSrcUid && gid < mntSrcGid {
```

Comment:
Same as above.

Reply:
Good catch here; for this case we do want a logical OR.